x-pack/plugins/observability_solution/synthetics/server/routes/settings/private_locations/migrate_legacy_private_locations.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import pMap from 'p-map';
+import { SavedObject } from '@kbn/core-saved-objects-server';
+import {
+  type PrivateLocationAttributes,
+  SyntheticsPrivateLocationsAttributes,
+} from '../../../runtime_types/private_locations';
+import {
+  legacyPrivateLocationsSavedObjectId,
+  legacyPrivateLocationsSavedObjectName,
+  privateLocationsSavedObjectName,
+} from '../../../../common/saved_objects/private_locations';
+import { RouteContext } from '../../types';
+
+export const migrateLegacyPrivateLocations = async ({
+  server,
+  request,
+  savedObjectsClient,
+}: RouteContext) => {
+  try {
+    const { coreStart } = server;
+    const canSave =
+      (
+        await coreStart?.capabilities.resolveCapabilities(request, {
+          capabilityPath: 'uptime.*',
+        })
+      ).uptime.save ?? false;
+
+    if (canSave) {
+      let obj: SavedObject<SyntheticsPrivateLocationsAttributes> | undefined;
+      try {
+        obj = await savedObjectsClient.get<SyntheticsPrivateLocationsAttributes>(
+          legacyPrivateLocationsSavedObjectName,
+          legacyPrivateLocationsSavedObjectId
+        );
+      } catch (e) {
+        return;
+      }
+      const legacyLocations = obj?.attributes.locations ?? [];
+      if (legacyLocations.length > 0) {
```

Comment:
```suggestion
      if (legacyLocations.length === 0 } {
         return;
      }
```

Add the return on if the locations are 0, that way you don't have to nest the below statement.